torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -3123,6 +3125,76 @@ Error PyTorchModelLoader::loadLogicalNot(const torch::jit::Node *ptNode) {
   return addValueMapping(outputs[0], F_.createNot("logical_not", glowInput));
 }
 
+Error PyTorchModelLoader::loadIndexPut(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+
+  glow::NodeValue input;
+  ASSIGN_VALUE_OR_RETURN_ERR(input, getGlowNodeValueForValue(inputs[0]));
+  std::vector<glow::NodeValue> *indices;
+  ASSIGN_VALUE_OR_RETURN_ERR(
+      indices, iValToNodeValueList(getGlowIValueForValue(inputs[1])));
+  glow::NodeValue value;
+  ASSIGN_VALUE_OR_RETURN_ERR(value, getGlowNodeValueForValue(inputs[2]));
+
+  auto *concatNode = F_.createConcat("concat", *indices, 0);
+  auto *reshapeNode =
+      F_.createReshape("reshape", concatNode->getResult(),
+                       {indices->size(), (*indices)[0].dims()[0]});
+  auto *transposeNode =
```

Comment:
Maybe we can first reshape each indices to a {1, dims[0], dims[1]...} tensor, and then concat them? Does this make our life easier?

Reply:
Replacing the current code with something like this also works

```
  std::vector<glow::NodeValue> reshapedIndices;
  for (auto &index : *indices) {
    reshapedIndices.push_back(
        F_.createExpandDims("expand_dims", index, {0})->getResult());
  }
  auto *concatNode = F_.createConcat("concat", reshapedIndices, 0);
  auto *transposeNode =
      F_.createTranspose("transpose", concatNode->getResult(), {1, 0});
```
but to me it seems to be about the same as the current code except it will create a reshape for each index tensor. Let me know if you prefer this approach or if I misunderstood what you meant.